### PR TITLE
Disable additional validation that is run post-build

### PR DESCRIPTION
**Pull request by mmitche**
Disable more validation stages that are run post-build. The only
thing left to run during each build at this point is SourceLink validation.
Once that is available, the entire validation stage should be able to be skipped

### CODE REVIEW

File: eng/pipelines/official/stages/publish.yml [hunk 1]
```diff
@@ -51,3 +46,14 @@ stages:
         -TsaRepositoryName "$(TsaRepositoryName)"
         -TsaCodebaseName "$(TsaCodebaseName)"
         -TsaPublish $True
+    # Allow symbol publish to emit expected warnings without failing the build. Include single
```
Comment (ViktorHofer) [inline eng/pipelines/official/stages/publish.yml:49]: What kind of expected warnings are we seeing during symbol publishing?

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
cc @safern

File: eng/pipelines/official/stages/publish.yml (hunk 1 above)
Comment (mmitche) [reply eng/pipelines/official/stages/publish.yml:49]: No idea...this was in there before.  I'll check some recent runs.

File: eng/pipelines/official/stages/publish.yml [hunk 2]
```diff
@@ -51,3 +46,14 @@ stages:
...
+    # quotes inside the string so that it passes through to MSBuild without script interference.
+    symbolPublishingAdditionalParameters: "'-warnAsError:$false'"
+    
+    # The following checks are not available after the build and continue to be run here
+    
+    # SourceLink validation doesn't work in dev builds: tries to pull from GitHub. https://github.com/dotnet/arcade/issues/3604
```
Comment (safern) [inline eng/pipelines/official/stages/publish.yml:55]: It seems like this comment can be removed.

**Review by safern — APPROVED** [review]

**Comment by mmitche** [comment]
No errors when I removed the -warnAsError:false.

**Comment by mmitche** [comment]
Performance tests completed and passed but just didn't update GH.

**Pull request merged**